docs/collecting-container-logs.md

```diff
@@ -0,0 +1,170 @@
+to appear in Sumo Logic as a single record, we need to tell the collector how to distinguish between lines which start a new record
+and ones which continue an existing record.
+
+Multiline log parsing can be configured using the `sumologic.logs.multiline` section in `user-values.yaml`.
+
+```yaml
+sumologic:
+  logs:
+    multiline:
+      enabled: true
+      first_line_regex: "^\\[?\\d{4}-\\d{1,2}-\\d{1,2}.\\d{2}:\\d{2}:\\d{2}"
+```
+
+where `first_line_regex` is a regular expression used to detect the first line of a multiline log.
+
+This feature is enabled by default. It can rarely cause problems by merging together lines which are supposed to be separate.
+In that case, feel free to disable it.
+
+### Setting source name and other built-in metadata
+
+It's possible to customize the built-in Sumo Logic metadata (like [source name][source_name] for example) for container logs:
+
+```yaml
+sumologic:
+  logs:
+    container:
+      ## Set the _sourceHost metadata field in Sumo Logic.
+      sourceHost: "%{k8s.pod.hostname}"
+      ## Set the _sourceName metadata field in Sumo Logic.
+      sourceName: "%{k8s.namespace.name}.%{k8s.pod.name}.%{k8s.container.name}"
+      ## Set the _sourceCategory metadata field in Sumo Logic.
+      sourceCategory: "%{k8s.namespace.name}/%{k8s.pod.pod_name}"
+      ## Set the prefix, for _sourceCategory metadata.
+      sourceCategoryPrefix: "kubernetes/"
+      ## Used to replace - with another character.
+      sourceCategoryReplaceDash: "/"
+```
+
+As can be seen in the above example, these fields can contain templates of the form `%{field_name}`, where `field_name` is the name
+of a resource attribute. Available resource attributes include [OpenTelemetry Kubernetes resource attributes][opentelemetry_k8s],
+in addition to the following:
+
+- `cluster`
+- `_collector`
+- `pod_labels_*` where * is the Pod label name
+
+### Filtering
+
+Logs can be excluded based on their container name, pod name, host, and namespace. This is done by providing a matching
+regular expression:
+
+```yaml
+sumologic:
+  logs:
+    container:
+      ## A regular expression for containers.
+      ## Matching containers will be excluded from Sumo. The logs will still be sent to logs metadata provider (otelcol).
+      excludeContainerRegex: ""
+      ## A regular expression for hosts.
+      ## Matching hosts will be excluded from Sumo. The logs will still be sent to logs metadata provider (otelcol).
+      excludeHostRegex: ""
+      ## A regular expression for namespaces.
+      ## Matching namespaces will be excluded from Sumo. The logs will still be sent to logs metadata provider (otelcol).
+      excludeNamespaceRegex: ""
+      ## A regular expression for pods.
+      ## Matching pods will be excluded from Sumo. The logs will still be sent to logs metadata provider (otelcol).
+      excludePodRegex: ""
+```
```

Comment:
Not really sure if we need this if we will have `Advanced filtering`

Reply:
I think it's needed, as you only need regex to use it, as opposed to otel configuration.

Reply:
I think it could be more generic, just list of pairs: attribute name, regex eg.

```
exclude:
  k8s.namespace.name: regexp
```

Reply:
I think we can add that later. This part is consistent with FluentD configuration and we have a migration that copies it over.